api/src/main/java/org/openmrs/api/impl/DiagnosisServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+		}
+		return dao.saveDiagnosis(diagnosis);
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#voidDiagnosis(Diagnosis, String)
+	 */
+	public Diagnosis voidDiagnosis(Diagnosis diagnosis, String voidReason) {
+		diagnosis.setVoided(true);
+		diagnosis.setVoidedBy(Context.getAuthenticatedUser());
+		diagnosis.setVoidReason(voidReason);
+		diagnosis.setDateVoided(new Date());
+		return save(diagnosis);
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnosisByUuid(String)
+	 */
+	public Diagnosis getDiagnosisByUuid(String uuid) {
+		return dao.getDiagnosisByUuid(uuid);
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnoses(Patient, Date)
+	 */
+	public List<Diagnosis> getDiagnoses(Patient patient, Date fromDate) {
+
+		EncounterSearchCriteria encounterSearchCriteria = new EncounterSearchCriteriaBuilder().setPatient(patient)
+			.setFromDate(fromDate).createEncounterSearchCriteria();
+		List<Encounter> encounters = Context.getEncounterService().getEncounters(encounterSearchCriteria);
+
+		List<Diagnosis> diagnoses = new ArrayList<Diagnosis>();
+
+		for (Encounter encounter : encounters) {
+			diagnoses.addAll(encounter.getDiagnoses());
+		}
+		return diagnoses;
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getPrimaryDiagnoses(Encounter)
+	 */
+	public List<Diagnosis> getPrimaryDiagnoses(Encounter encounter) {
+		List<Diagnosis> diagnoses = null;
+		if (encounter != null && !encounter.getVoided()) {
+			diagnoses = new ArrayList<Diagnosis>();
+
+			for (Diagnosis diagnosis :
+				encounter.getDiagnoses()) {
+				// check if the diagnosis is given the highiest importance. 
+				if (diagnosis.getOrder == DIAGNOSIS_ORDER_PRIMARY) {
+					diagnoses.add(diagnosis);
+				}
+			}
+		}
+		return diagnoses;
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#hasDiagnosis(Encounter, Diagnosis)
+	 */
+	public boolean hasDiagnosis(Encounter encounter, Diagnosis diagnosis) {
+
+		for (Diagnosis diagnosisFound :
+			encounter.getDiagnoses()) {
+			if (diagnosis == diagnosisFound) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getUniqueDiagnoses(Patient, Date)
+	 */
+	public List<Diagnosis> getUniqueDiagnoses(Patient patient, Date fromDate) {
+		List<Diagnosis> diagnoses = getDiagnoses(patient, fromDate);
+
+		Set<CodedOrFreeText> uniqueDiagnoses = new HashSet<CodedOrFreeText>();
+		Iterator<Diagnosis> it = diagnoses.iterator();
+
+		while (it.hasNext()) {
+			Diagnosis diagnosis = it.next();
+			if (!uniqueDiagnoses.add(diagnosis.getDiagnosis())) {
+				it.remove();
+			}
+		}
+		return diagnoses;
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#unvoidDiagnosis(Diagnosis)
+	 */
+	public Diagnosis unvoidDiagnosis(Diagnosis diagnosis) throws APIException {
+		if (diagnosis.getVoided()) {
+			diagnosis.setVoided(false);
+			diagnosis.setVoidedBy(null);
+			diagnosis.setDateVoided(null);
+			diagnosis.setVoidReason(null);
+		}
+		return save(diagnosis);
```

Comment:
Same here, you don't need to set anything here and you need to get the service from the Context and call save on that